uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -1440,4 +1439,47 @@ private static function getCurrentAbsenceType($leaveRequestID) {
 
     return AbsenceType::findById($currentLeaveRequest->type_id);
   }
+
+  /**
+   * Get the number of working days for a leave request, Days with
+   * balance change of zero or that evaluates to false are either non
+   * working day, weekend or public holidays.
+   * The days for which the balance amounts are not zero are equal
+   * to the number of working days.
+   *
+   * @param array $balanceDetails
+   *
+   * @return int
+   */
+  private static function getNumberOfWorkingDaysForLeaveRequest($balanceDetails) {
+    $balanceAmounts = array_column($balanceDetails['breakdown'], 'amount');
+    $balanceAmounts = array_filter($balanceAmounts);
+
+    return count($balanceAmounts);
+  }
+
+  /**
+   * Returns the balance change breakdown for a leave request
+   * based on the parameters passed to LeaveRequest.create.
+   *
+   * @param array $params
+   *
+   * @return array
+   */
+  private static function getLeaveBalanceBreakdown($params) {
```

Comment:
This method ended up being confusing because the Breakdown is just part of what the `calculateBalanceChange()` method returns, but `getLeaveBalanceBreakdown()` returns exactly the same thing as `calculateBalanceChange()`